js/src/setup-mc/setup-stepper/store-requirements/index.js

```diff
@@ -36,10 +36,26 @@ export default function StoreRequirements() {
 	 * all onboarding accounts are considered unverified phone numbers.
 	 */
 	const [ isPhoneNumberReady, setPhoneNumberReady ] = useState( false );
+	const [ isSaveSettings, setSaveSettings ] = useState( true );
 	const [ completing, setCompleting ] = useState( false );
 
-	const handleChangeCallback = ( _, values ) => {
-		saveSettings( values );
+	const handleChangeCallback = async ( _, values ) => {
+		try {
+			await saveSettings( values );
+			setSaveSettings( true );
+		} catch ( error ) {
+			//Create the notice only once
+			if ( isSaveSettings === true ) {
+				createNotice(
+					'error',
+					__(
+						'There was an error trying to save settings. Please try again later.',
+						'google-listings-and-ads'
+					)
+				);
```

Comment:
Thanks for this suggestion, I have tried and I saw the following points: 

1. This needs to be created as react hook. The `createNotice` depends on `useDispatchCoreNotices`.
2. The react hook will need to return a function wrapping the createNotice, so it can be called anywhere in code, otherwise it will complain about it:

`React Hook "useErrorSavingSettingsNotice" is called in function "saveSettingsCallback" that is neither a React function component nor a custom React Hook function`

So I think adding this hook to create the notice will create unnecessary complexity. WDYT